libs/processors/src/main/java/org/wordpress/android/processor/RemoteConfigProcessor.kt

```diff
@@ -1,128 +1,128 @@
-import javax.annotation.processing.RoundEnvironment
-import javax.annotation.processing.SupportedAnnotationTypes
-import javax.annotation.processing.SupportedSourceVersion
-import javax.lang.model.SourceVersion
-import javax.lang.model.element.TypeElement
-import javax.tools.Diagnostic.Kind
-
-@AutoService(Processor::class) // For registering the service
-@SupportedSourceVersion(SourceVersion.RELEASE_8) // to support Java 8
-@SupportedAnnotationTypes(
-    "org.wordpress.android.annotation.Experiment",
-    "org.wordpress.android.annotation.Feature",
-    "org.wordpress.android.annotation.FeatureInDevelopment",
-    "org.wordpress.android.annotation.RemoteFieldDefaultGenerater"
-)
-class RemoteConfigProcessor : AbstractProcessor() {
-    @OptIn(DelicateKotlinPoetApi::class)
-    @Suppress("DEPRECATION")
-    override fun process(p0: MutableSet<out TypeElement>?, roundEnvironment: RoundEnvironment?): Boolean {
-        val experiments = roundEnvironment?.getElementsAnnotatedWith(Experiment::class.java)?.map { element ->
-            val annotation = element.getAnnotation(Experiment::class.java)
-            annotation.remoteField to annotation.defaultVariant
-        } ?: listOf()
-        val remoteFeatureNames = mutableListOf<TypeName>()
-        val features = roundEnvironment?.getElementsAnnotatedWith(Feature::class.java)?.map { element ->
-            val annotation = element.getAnnotation(Feature::class.java)
-            remoteFeatureNames.add(element.asType().asTypeName())
-            annotation.remoteField to annotation.defaultValue.toString()
-        } ?: listOf()
-        val remoteFields = roundEnvironment?.getElementsAnnotatedWith(RemoteFieldDefaultGenerater::class.java)
-            ?.map { element ->
-                val annotation = element.getAnnotation(RemoteFieldDefaultGenerater::class.java)
-                annotation.remoteField to annotation.defaultValue
-            } ?: listOf()
-        val featuresInDevelopment = roundEnvironment?.getElementsAnnotatedWith(FeatureInDevelopment::class.java)
-            ?.map { element ->
-                element.asType().toString()
-            } ?: listOf()
-        return if (experiments.isNotEmpty() || features.isNotEmpty()) {
-            generateRemoteFieldConfigDefaults(remoteFields.toMap())
-            generateRemoteFeatureConfigDefaults((experiments + features).toMap())
-            generateRemoteFeatureConfigCheck(remoteFeatureNames)
-            generateFeaturesInDevelopment(featuresInDevelopment)
-            true
-        } else {
-            false
+
+@OptIn(KspExperimental::class)
+class RemoteConfigProcessor(
+    private val codeGenerator: CodeGenerator,
+) : SymbolProcessor {
+    /**
+     * See: https://github.com/google/ksp/issues/797#issuecomment-1041127747
+     * Also: https://github.com/google/ksp/blob/a0cd7774a7f65cec45a50ecc8960ef5e4d47fc21/examples/playground/test-processor/src/main/kotlin/TestProcessor.kt#L20
+     */
+    private var invoked = false
```

Comment:
In the case of this processor, we only one need round - there generated files do not depend on each other or any other processors.

Reply:
Sure! Added in 0292847e39d9e704b117c6f3c53fb494881ad6a1